### ISSUE TITLE
Improve accessibility of the generated sites

### ISSUE THREAD

**Issue by ralsina** [getnikola/nikola#377]
Creating accessible sites is very important, and there are many ways in which Nikola can improve in the area.

Examples:

* The search form has no label for the search field
* Figures could use caption as ALT instead of the filename
* Skip to content link could be added

And lots more, taken from a report by http://achecker.ca/

Let's consider this a "master issue" for accessibility.

**Comment by ralsina** [comment]
Currently achecked.ca shows no issues with Nikola's output according to WCAG 2.0 (level AA). Is that good enough?

**Comment by Aeyoun** [comment]
http://www.w3.org/WAI/PF/aria-practices/

**Comment by Kwpolska** [comment]
Perhaps the best way to find out would be to ask a vision impaired user.
There is a Nikola user that could help, his name is Mike Ray of the Eyes
Free Linux Ninja blog. Should I email him and ask for help?

(His blog uses the base theme; he says it's good.)

http://www.w3.org/WAI/PF/aria-practices/

—
Reply to this email directly or view it on GitHub
<https://github.com/getnikola/nikola/issues/377#issuecomment-101233144>.

**Comment by ralsina** [comment]
On 12/05/15 16:49, Chris Warrick wrote:
> Perhaps the best way to find out would be to ask a vision impaired user.
> There is a Nikola user that could help, his name is Mike Ray of the Eyes
> Free Linux Ninja blog. Should I email him and ask for help?
>
> (His blog uses the base theme; he says it's good.)
>

Sounds good, ask him please!

**Comment by Kwpolska** [comment]
Mail sent.

One thing we are not doing that is recommended by the Bootstrap 3 documentation is:

```html
        <li><a href="#">Link</a></li>
        <li class="active"><a href="#">Link <span class="sr-only">(current)</span></a></li>
```

**Comment by ralsina** [comment]
Should be easy-ish

**Comment by Kwpolska** [comment]
We can add it pretty easily, we just need to take care of i18n.

**Comment by Kwpolska** [comment]
Preliminary input: it’s looking pretty good.  The only thing is, dropdowns are not working (but that’s not our fault).  I also had to rewrite the demo post not to include “in order to foo a bar, [click here](#)” links (which are not accessible if you are navigating by links — “[foo a bar](#)” is a much better way to put it anyway)

I should get further input soon and will post everything and handle the fixes (`a11y` branch).
